common.py

```diff
@@ -643,25 +643,90 @@ def cut(self, xmin=-np.inf, xmax=np.inf):
         h.errs = h.errs[imin:imax-1]
         return h
 
+def _create_binning(binw, left, right):
+    return left + binw * np.arange(math.ceil((right-left)/binw)+1)
 
-class MTHistogram(Histogram):
+class VarArrHistogram(Histogram):
     """
-    Small wrapper around Histogram that initializes from mt values and weights.
+    Wrapper around histogram that initializes the value and weight arrays.
+    It will also carry around the current and default binning information
     """
+    default_binning = (10,0,100) # Width, min, max
+    bins = None # List of bin values
+    name = '' # Name of variable to use
+
+    @property
+    def non_standard_binning(self):
+        if self.bins is None:
+            self.bins = self.create_binning(*self.default_binning)
+        if len(self.bins) != self.create_binning(*self.default_binning):
+            return False
+        if not all(np.isclose(self.bins, self.create_binning(*self.default_binning))):
+            return False
+        return True
+
+    @classmethod
+    def default_binw(cls)->float:
+        return cls.default_binning[0]
+
+    @classmethod
+    def default_binmin(cls)->float:
+        return cls.default_binning[1]
+
+    @classmethod
+    def default_binmax(cls)->float:
+        return cls.default_binning[2]
+
+    @classmethod
+    def create_binning(cls, binw, left, right):
+        return _create_binning(binw, left, right)
 
-    bins = MT_BINS.copy()
-    non_standard_binning = False
 
     @classmethod
     def empty(cls):
         return Histogram(cls.bins)
 
-    def __init__(self, mt, weights=None):
-        vals = np.histogram(mt, self.bins, weights=weights)[0].astype(float)
-        errs = np.sqrt(np.histogram(mt, self.bins, weights=weights**2)[0].astype(float))
+    def __init__(self, cols, weights=None):
+        if self.bins is None:
+            self.bins = self.create_binning(*self.default_binning)
+        var_arr = cols.to_numpy([self.name]).flatten()
```

Comment:
should eventually become a function so that a class can override it to implement a more involved calculation